pkg/sql/internal_test.go

```diff
@@ -106,6 +107,41 @@ func TestInternalExecutor(t *testing.T) {
 	}
 }
 
+func TestInternalFullTableScan(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+	params, _ := tests.CreateTestServerParams()
+	s, db, _ := serverutils.StartServer(t, params)
+	defer s.Stopper().Stop(ctx)
+
+	_, err := db.Exec("CREATE DATABASE db; SET DATABASE = db;")
+	require.NoError(t, err)
+
+	_, err = db.Exec("CREATE TABLE t(a INT)")
+	require.NoError(t, err)
+
+	_, err = db.Exec("INSERT INTO t VALUES (1), (2), (3)")
+	require.NoError(t, err)
+
+	_, err = db.Exec("SET disallow_full_table_scans = true")
```

Comment:
I'm not convinced that this is testing anything. This is a session setting, and accessing the ie directly through s wouldn't inherit those settings. Does this fail if you comment out your guard logic on the error path?

A better example would be selecting from `crdb_internal.jobs` or something that performs a full table scan through an IE that is actually part of the current session